### PR TITLE
added sample code for fasterrcnn_resnet50_fpn (optional)

### PR DESCRIPTION
**Pull request by prajjwal1**
Discussed [here](https://github.com/pytorch/vision/issues/1688).

### CODE REVIEW

File: intermediate_source/torchvision_tutorial.rst [hunk 1]
```diff
@@ -304,6 +304,30 @@ be using Mask R-CNN:
 That’s it, this will make ``model`` be ready to be trained and evaluated
 on your custom dataset.
 
+Checking the model with random tensors (Optional)
+---------------------------
+
+Before iterating over the dataset, it's always good to see what the model 
+expects during training and inference time with random tensors.
```
Comment (fmassa) [inline intermediate_source/torchvision_tutorial.rst]: This is something we should be careful about: if we have batch normalization layers in the model (which is not the case for detection models because we freeze the batch norm layers, but happens in classification), feeding randomly-initialized data will affect the batch norm statistics, leading to potentially unexpected results

File: intermediate_source/torchvision_tutorial.rst [hunk 2]
```diff
@@ -304,6 +304,30 @@ be using Mask R-CNN:
...
+.. code:: python
+
+   model = torchvision.models.detection.fasterrcnn_resnet50_fpn(pretrained=True) 
+   images,boxes,labels = torch.rand(4,3,600,1200), torch.rand(4,11,4), torch.rand(4,11) # For Training
+   images = list(image for image in images)                                             
+   targets = []                                   
+   for i in range(len(images)):
+       d = {}
+       d['boxes'] = boxes[i]
+       d['labels'] = labels[i].type(torch.int64)
```
Comment (fmassa) [inline intermediate_source/torchvision_tutorial.rst]: The use of `.type` is not encouraged (it's legacy), and instead we should use `.to`. See my fixes (including lint fixes) in https://github.com/pytorch/vision/pull/1713

File: intermediate_source/torchvision_tutorial.rst (hunk 1 above)
Comment (prajjwal1) [reply intermediate_source/torchvision_tutorial.rst]: I've now changed the code from using random tensors to sampled data from the actual dataset. This will fix the batch norm stats issue.

File: intermediate_source/torchvision_tutorial.rst (hunk 2 above)
Comment (prajjwal1) [reply intermediate_source/torchvision_tutorial.rst]: Thanks for pointing this out. I've fixed it.

**Comment by prajjwal1** [comment]
Let me know, if this seems fine.

**Comment by prajjwal1** [comment]
Thanks, what's the next step now?

**Comment by prajjwal1** [comment]
@brianjo Can you please merge it ?

**Comment by brianjo** [comment]
> @brianjo Can you please merge it ?

Hi @prajjwal1, @yf225 and I are working on fixing the build right now and I'll rebase and merge this one when we get that going. Thanks!

**Comment by prajjwal1** [comment]
@yf225 , Can you please have a look at this ? Can this be merged now. I'll close the issue in torchvision if it's merged.

**Comment by yf225** [comment]
@prajjwal1 I believe all build jobs passed on this PR. @brianjo Do you think we can merge this PR?

**Comment by brianjo** [comment]
> @prajjwal1 I believe all build jobs passed on this PR. @brianjo Do you think we can merge this PR?

Will do. Still working through a Netlify issue, but this looks good.

**Comment by prajjwal1** [comment]
Thanks !
